agents/kubevirt/fence_kubevirt.py

```diff
@@ -0,0 +1,123 @@
+        vm_list = vm_api.get(namespace=namespace)
+        for vm in vm_list.items:
+            result[vm.metadata.name] = ("", None)
+    except Exception as e:
+        logger.error("Exception when calling VirtualMachine list: %s", e)
+    return result
+
+def get_power_status(conn, options):
+    logger.debug("Starting get status operation")
+    try:
+        namespace = options.get("--namespace")
+        name = options.get("--plug")
+        vmi_api = conn.resources.get(api_version=API_VERSION,
+                                              kind='VirtualMachineInstance')
+        vmi = vmi_api.get(name=name, namespace=namespace)
+        if vmi is not None:
+            phase = vmi.status.phase
+            if phase == "Running":
+                return "on"
+        return "off"
+    except ApiException as e:
+        if e.status == 404:
+            return "off"
+        logging.error("Failed to get power status, with API Exception: %s", e)
+        fail(EC_STATUS)
+    except Exception as e:
+        logging.error("Failed to get power status, with Exception: %s", e)
+        fail(EC_STATUS)
+
+def set_power_status(conn, options):
+    logger.debug("Starting set status operation")
+    try:
+        namespace = options.get("--namespace")
+        name = options.get("--plug")
+        action = 'start' if options["--action"] == "on" else 'stop'
+        virtctl_vm_action(conn, action, namespace, name)
+    except Exception as e:
+        logging.error("Failed to set power status, with Exception: %s", e)
+        fail(EC_STATUS)
+
+def define_new_opts():
+	all_opt["namespace"] = {
+		"getopt" : ":",
+		"longopt" : "namespace",
+		"help" : "--namespace=[namespace]        Namespace",
+		"shortdesc" : "Namespace.",
+		"required" : "1",
+		"order" : 2
+	}
+	all_opt["kubeconfig"] = {
+		"getopt" : "k:",
+		"longopt" : "kubeconfig",
+		"help" : "-k, --kubeconfig=[kubeconfig]  Kubeconfig file path",
+		"shortdesc": "Kubeconfig file path",
+		"required": "0",
+		"order": 4
+	}
+
+def virtctl_vm_action(conn, action, namespace, name):
+    path = '/apis/subresources.{api_version}/namespaces/{namespace}/virtualmachines/{name}/{action}'
+    path = path.format(api_version=API_VERSION, namespace=namespace, name=name, action=action)
+    return conn.request('put', path, has_accept_params=False)
+
+def validate_options(required_options_list, options):
+    for required_option in required_options_list:
+        if required_option not in options:
+            fail_usage("Failed: %s option must be provided" % required_option)
+
+# Main agent method
+def main():
+    conn = None
+
+    device_opt = ["port", "namespace", "kubeconfig", "separator", "no_password"]
+    define_new_opts()
+    options = check_input(device_opt, process_input(device_opt))
+
+    docs = {}
+    docs["shortdesc"] = "Fence agent for Kubevirt"
+    docs["longdesc"] = "fence_kubevirt is an I/O Fencing agent for Kubevirt."
```

Comment:
I think this agent does not fence I/O, it kills the compute. Am I wrong?

Reply:
This is just what agents say in general, so it's fine.